tests/loki/loki_iofq_test.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from typing import Callable, List
+
+import numpy as np
+import pytest
+import sciline
+import scipp as sc
+from loki_common import make_params
+
+import esssans as sans
+from esssans.types import (
+    BackgroundSubtractedIofQ,
+    FinalDims,
+    UncertaintyBroadcastMode,
+    WavelengthBands,
+)
+
+
+def loki_providers() -> List[Callable]:
+    return list(sans.providers + sans.loki.providers)
+
+
+def test_can_create_pipeline():
+    pipeline = sciline.Pipeline(loki_providers(), params=make_params())
+    pipeline.get(BackgroundSubtractedIofQ)
+
+
+@pytest.mark.parametrize(
+    'uncertainties',
+    [UncertaintyBroadcastMode.drop, UncertaintyBroadcastMode.upper_bound],
+)
+def test_pipeline_can_compute_IofQ(uncertainties):
+    params = make_params()
+    params[UncertaintyBroadcastMode] = uncertainties
+    pipeline = sciline.Pipeline(loki_providers(), params=params)
+    result = pipeline.compute(BackgroundSubtractedIofQ)
+    assert result.dims == ('Q',)
+
+
+def test_pipeline_can_compute_IofQ_in_wavelength_slices():
+    params = make_params()
+    band = np.linspace(1.0, 13.0, num=11)
+    params[WavelengthBands] = sc.array(
+        dims=['band', 'wavelength'],
+        values=np.vstack([band[:-1], band[1:]]).T,
+        unit='angstrom',
+    )
+    pipeline = sciline.Pipeline(loki_providers(), params=params)
+    result = pipeline.compute(BackgroundSubtractedIofQ)
+    assert result.dims == ('band', 'Q')
```

Comment:
`FinalDims` says `('Q',)`, so this is a bit odd. Is the parameter badly named? Should we instead list which dims not to reduce, or something like that?

Reply:
> Should we instead list which dims not to reduce, or something like that?

Yeah, I came up with that `FinalDims` params at some point but never really liked it.
It was a way to be flexible, in case we wanted to return an `I(Q)` per straw or anything we want.
However, that use case is a bit unlikely? We would mostly just want it per layer.

We could either do what you suggest, or we could just have a boolean param that is `perLayer = True/False`?